src/fstests_runner.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python3
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# Copyright (C) 2022 Collabora Limited
+# Author: Alexandra Pereira <alexandra.pereira@collabora.com>
+
+import logging
+import os
+import sys
+
+import kernelci
+import kernelci.config
+import kernelci.db
+import kernelci.lab
+from kernelci.cli import Args, Command, parse_opts
+
+from logger import Logger
+
+
+class FstestsRunner:
+    def __init__(self, configs, args):
+        self._db_config = configs['db_configs'][args.db_config]
+        api_token = os.getenv('API_TOKEN')
+        self._db = kernelci.db.get_db(self._db_config, api_token)
+        self._logger = Logger("config/logger.conf", "complete")
+        self._tarball_node = None
+
+    def run(self):
+        sub_id = self._db.subscribe_node_channel({
+            'name': 'tarball',
+        })
+        self._logger.log_message(logging.INFO, "Listening for tarballs")
+        self._logger.log_message(logging.INFO, "Press Ctrl-C to stop.")
+
+        try:
+            while True:
+                if not self._tarball_node:
+                    self._tarball_node = self._db.receive_node(sub_id)
```

Comment:
This should be a local variable rather than an object member:
```suggestion
                    tarball_node_id = self._db.receive_node(sub_id)
```

Reply:
updated